### ISSUE TITLE
Add possibilityto pass arbitrary keys engine.setValue and engine.setParameter 

### ISSUE THREAD

**Issue by christophehenry** [mixxxdj/mixxx#12766]
### Feature Description

First things first, sorry if there's already an issue opened for this feature. I tried my best to find one but maybe I didn't search for the right words. 

It is not rare for a device to have component that modifies other components behavior.

For instance, the Denon DN-S3700 has a Break button that corresponds to the ` engine.brake()` function. It behaves like this:

1. hit the break button,
2. turn the parameter knob to select the break duration,
3. hit the parameter knob's button to enable the brake effect,
4. during the playback, hit the play button to pause with a brake effect.

Implementing this feature requires 3 components for setting the effect and modify the behavior of a 4th.

While all of this can be implemented with properties and functions in JS, retaining state with arbitrary Mixxx controls enable reactive programming, which simplifies the code and is less error prone.

This could be implemented by introducing a `[CustomX]` control group. For instance: `engine.setValue("[Custom1]", "playState", 1)`°

**Comment by Swiftb0y** [comment]
This has been requested often, but it has lots of implications that we want to avoid. Reactive programming makes definitely sense, but it would be better to implement natively by writing mappings in QML (currently not possible, but wanted).

**Comment by christophehenry** [comment]
Is it related in some way to the *[Registering MIDI Input Handlers From Javascript*](https://github.com/mixxxdj/mixxx/wiki/registering-midi-input-handlers-from-javascript) topic in some way?

**Comment by Swiftb0y** [comment]
Not really, that is separate. It's related rather to the fact that arbitrary COs for controller mappings (which have been requested previously for inter-mapping communication) are bound to introduce compatibility issues between mappings.
(also just fyi since you're currently learning the controller mapping system, prefer to use `engine.makeConnection` over `engine.connectControl`, the latter has been deprecated for quite a while).

**Comment by christophehenry** [comment]
COs? Ok, so how hard would that be? Where should I start if I wanted to work on this?

**Comment by Swiftb0y** [comment]
> COs?

Sorry for glossing over that. That short for what we call "Control Objects" and its the `[group],key` API you're interacting with via the `engine.set*` calls. 

> Ok, so how hard would that be?

Medium I'd guess. Like 50 SLOC. But as I said, that is not a good solution and I would be inclined to reject such a PR currently. 

---

I want to help you, so lets get back to the drawing board. What is your current usecase? I assume the break example you posted above?

**Comment by christophehenry** [comment]
Some of it, yes. The DN-S3700 implements several features like this one, involving multiple components interacting with one another. In the abstract, this is entirely solvable with states in JS variables. The reactive programming that Mixxx controls offers just makes the code cleaner and I was interested in improving Mixxx anyway.

**Comment by Swiftb0y** [comment]
> The reactive programming that Mixxx controls offers just makes the code cleaner and I was interested in improving Mixxx anyway.

I agree, but connections to COs is not the solution. For now, regular imperative OO-oriented programming suffices for controller mappings (I recommend you take a look at ComponentJS).

If you want to improve mixxx controller scripting, I recommend you take a look at some small features such as the registering input handlers project you already discovered. Those will likely take a medium amount of code (300-500 SLOC I'm guessing) and would already help a lot in making the mapping system better.
